tests/e2e/gauntlet_deployer.go

```diff
@@ -0,0 +1,370 @@
+	Expect(err).ShouldNot(HaveOccurred(), "Failed to deploy the deviation flagging validator")
+	dfvReport, err := LoadReportJson(reportName + ".json")
+	Expect(err).ShouldNot(HaveOccurred())
+	return GetTxAddressFromReport(dfvReport)
+}
+
+// DeployOcr deploys ocr, it creates an rdd file in the process and updates it with the ocr address on completion
+func (gd *GauntletDeployer) DeployOcr() (string, string) {
+	rddPath := filepath.Join("tests", "e2e", "smoke", "rdd", fmt.Sprintf("directory-terra-%s.json", gd.Cli.Network))
+	tmpId := "terra1test0000000000000000000000000000000000"
+	ocrRddContract := NewRddContract(tmpId)
+	err := WriteRdd(ocrRddContract, rddPath)
+	Expect(err).ShouldNot(HaveOccurred(), "Did not write the rdd json correctly")
+	reportName := "ocr_deploy"
+	UpdateReportName(reportName, gd.Cli)
+	_, err = gd.Cli.ExecCommandWithRetries([]string{
+		"ocr2:deploy",
+		gd.Cli.Flag("rdd", rddPath),
+		gd.Cli.Flag("version", gd.Version),
+		tmpId,
+	}, []string{
+		TERRA_COMMAND_ERROR,
+	}, RETRY_COUNT)
+	Expect(err).ShouldNot(HaveOccurred(), "Failed to deploy ocr2")
+	ocrReport, err := LoadReportJson(reportName + ".json")
+	Expect(err).ShouldNot(HaveOccurred())
+	ocr := GetTxAddressFromReport(ocrReport)
+
+	// add the new contract to the rdd
+	ocrRddContract.Contracts[ocr] = ocrRddContract.Contracts[tmpId]
+	err = WriteRdd(ocrRddContract, rddPath)
+	Expect(err).ShouldNot(HaveOccurred(), "Did not write the rdd json correctly")
+	return ocr, rddPath
+}
+
+// SetBiling sets the billing info that exists in the rdd file for the ocr address you pass in
+func (gd *GauntletDeployer) SetBilling(ocr, rddPath string) {
+	UpdateReportName("set_billing", gd.Cli)
+	_, err := gd.Cli.ExecCommandWithRetries([]string{
+		"ocr2:set_billing",
+		gd.Cli.Flag("version", gd.Version),
+		gd.Cli.Flag("rdd", rddPath),
+		ocr,
+	}, []string{
+		TERRA_COMMAND_ERROR,
+	}, RETRY_COUNT)
+	Expect(err).ShouldNot(HaveOccurred(), "Failed to set billing")
+}
+
+// BeginProposal begins the proposal
+func (gd *GauntletDeployer) BeginProposal(ocr, rddPath string) string {
+	reportName := "begin_proposal"
+	UpdateReportName(reportName, gd.Cli)
+	_, err := gd.Cli.ExecCommandWithRetries([]string{
+		"ocr2:begin_proposal",
+		gd.Cli.Flag("version", gd.Version),
+		gd.Cli.Flag("rdd", rddPath),
+		ocr,
+	}, []string{
+		TERRA_COMMAND_ERROR,
+	}, RETRY_COUNT)
+	Expect(err).ShouldNot(HaveOccurred(), "Failed to begin proposal")
+	beginProposalReport, err := LoadReportJson(reportName + ".json")
+	Expect(err).ShouldNot(HaveOccurred())
+	return beginProposalReport["data"].(map[string]interface{})["proposalId"].(string)
+}
+
+// ProposeConfig proposes the config
+func (gd *GauntletDeployer) ProposeConfig(ocr, proposalId, rddPath string) {
+	reportName := "propose_config"
+	UpdateReportName(reportName, gd.Cli)
+	_, err := gd.Cli.ExecCommandWithRetries([]string{
+		"ocr2:propose_config",
+		gd.Cli.Flag("version", gd.Version),
+		gd.Cli.Flag("rdd", rddPath),
+		gd.Cli.Flag("f", "1"),
```

Comment:
`f` comes from RDD, so this shouldn't be necessary

Reply:
fixed